openupgrade_scripts/scripts/account/14.0.1.1/post-migration.py

```diff
@@ -605,6 +605,69 @@ def fill_account_journal_payment_credit_debit_account_id(env):
         )
 
 
+def create_new_counterpar_account_payment_transfer(env):
+    # Create new counterpart payment with account payment transfer
+    openupgrade.logged_query(
+        env.cr,
+        """
+        INSERT INTO account_payment (move_id, is_internal_transfer, partner_type,
+            payment_type,
+            amount, currency_id,
+            destination_account_id, partner_id, journal_id,
+            create_uid, create_date, write_uid, write_date)
+        SELECT move.id, true, ap.partner_type,
+            CASE
+            WHEN liquidity_line.amount_currency > 0 THEN 'inbound' ELSE 'outbound'
+            END,
+            ABS(liquidity_line.amount_currency), liquidity_line.currency_id,
+            ap.destination_account_id, ap.partner_id, move.journal_id,
+            move.create_uid, move.create_date, move.write_uid, move.write_date
+        FROM account_payment ap
+        JOIN account_move move
+            ON (move.payment_id = ap.id AND move.id != ap.move_id)
+        JOIN account_journal journal ON journal.id = move.journal_id
+        JOIN account_move_line liquidity_line
+            ON liquidity_line.move_id = move.id
+                AND liquidity_line.account_id IN (
+                    journal.default_account_id,
+                    journal.payment_debit_account_id,
+                    journal.payment_credit_account_id
+                    )
+        WHERE ap.payment_type = 'transfer'
+        """,
+    )
+
+
+def map_account_payment_transfer(env):
+    # map payment_type from transfer to 'inbound'/'outbound'
+    # and set is_internal_transfer as true on account payment transfer
+    openupgrade.logged_query(
+        env.cr,
+        """
+        WITH subquery AS (
+            SELECT ap.id, line.amount_currency
+            FROM account_payment ap
+            JOIN account_journal journal ON journal.id = ap.journal_id
+            LEFT JOIN account_move_line line
+                ON line.move_id = ap.move_id
+                    AND line.account_id IN (
+                        journal.default_account_id,
+                        journal.payment_debit_account_id,
+                        journal.payment_credit_account_id
+                        )
+            WHERE ap.payment_type = 'transfer'
+        )
```

Comment:
Unneeded. See my final explanation.

Reply:
Still need this part, because we just set to True for the new payment, the old payment has not been set yet

Reply:
Yeah, but I'm talking about the subquery and all of that. Simply those that have 2 journals are internal.